pandas/io/formats/style.py

```diff
@@ -561,11 +561,15 @@ def _update_ctx(self, attrs: DataFrame) -> None:
             Whitespace shouldn't matter and the final trailing ';' shouldn't
             matter.
         """
-        for row_label, v in attrs.iterrows():
-            for col_label, col in v.items():
-                i = self.index.get_indexer([row_label])[0]
-                j = self.columns.get_indexer([col_label])[0]
-                for pair in col.rstrip(";").split(";"):
+        rows = [(row_label, v) for row_label, v in attrs.iterrows()]
```

Comment:
you should use itertuples here (its actually much faster)

Reply:
@Jeff - that's a good thing to know and I tried it but could not figure out doing the same thing with itertuples.
To get the col_label within the inner loop, I need to use ._fields(), getattr(), list slicing, etc to separate index, ... basically many extra steps. I am not sure how much we can save here.

However, it seems that .get_indexer is the one that caused much delay. So real solution should be something that will eliminate get_indexer entirely or some acceleration effort done on get_indexer.

I can think of one way to avoid get_indexer -- simply taking index & columns as list and use it to get integer index # of given label. However, I was not sure if I could do that safely because I am not sure all the labels given in attrs always matches that of self.index and self.columns. probably not.

Reply:
ahh i see, you are doing get indexer once for the rows, you can do the same once for the columns. you can throw this in a dict {label -> int}. This will vastly speed up things.

Reply:
So you mean, each row will have same columns ..?
how about 4x4 table that has some attr assignments on column A, B on row 1, 2 but on column C, D only on row 3, 4 ..?
It would be great if the original author of this function could answer this -- or, are you may be?
I admit that I did the patch much relying on guesses based on common sense ( or my version of common sense :) )

Reply:
If it is certain that we really do not have to use get_indexer method, probably something this should work, outside the loops:
rowmap = { label: i for i, label in enumerate(self.index) }
colmap = { label: i for i, label in enumerate(self.columns) }

Reply:
This is the exact code that worked for my app but not sure if it will work in all occations:
```
    def _update_ctx(self, attrs: DataFrame) -> None:
        """
        Update the state of the Styler.

        Collects a mapping of {index_label: ['<property>: <value>']}.

        Parameters
        ----------
        attrs : DataFrame
            should contain strings of '<property>: <value>;<prop2>: <val2>'
            Whitespace shouldn't matter and the final trailing ';' shouldn't
            matter.
        """
        cols = list(self.columns)
        rows = list(self.index)
        for ii in range(len(attrs.columns)):
            cn = attrs.columns[ii]
            i = cols.index(cn)
            for ri, c in attrs[[cn]].itertuples():
                if not c: continue
                c = c.rstrip(";")
                if not c: continue
                j = rows.index(ri)
                for pair in c.split(";"):
                    self.ctx[(j, i)].append(pair)
```

Reply:
can you avoid the append? I think if this was a comprehension (or at least the last append) would be much better

Reply:
I don't see how how it can be better.
Let's say we have a defaultdict out of comprehension with new pairs to be added to existing ctx. What should happen afterward is basically same things as this one, I would assume.

Reply:
ok i actually like your code above a little better, its very idiomatic and easy to understand. push it up and ping on green.

Reply:
I really don't feel safe with this code. It might break someone's code.
What if the attrs's column is specified different way other than plain column label?